### ISSUE TITLE
deprecatedModuleAttribute emits warning twice when deprecated module is imported

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4492]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)| @cyli reported|
|-|-|
|Trac ID|trac#4492|
|Type|defect|
|Created|2010-06-05 23:47:36Z|
|Branch|https://github.com/twisted/twisted/tree/deprecated-module-4492|

See ticket [[#4096](https://github.com/twisted/twisted/issues/4096)](https://github.com/twisted/twisted/issues/4096), in particular [http://twistedmatrix.com/trac/attachment/ticket/4096/4096.diff].

When a module is deprecated in the parent module, flushWarnings() will produce 2 duplicate warnings when said deprecated module is imported.

"Because of the way imports from packages work, the deprecation is first emitted by an insults.colors(-like) access which raises an AttributeError. The access is then retried after colors.py has actually been loaded; the warning is emitted again, and this time the lookup succeeds and the import is complete. Normally this will probably be invisible, because duplicate warnings from the same location are suppressed by the reporting code, but trial goes out of its way to make sure every warning gets emitted."

(insults.colors was deprecated in #4096 by using `deprecatedModuleAttribute` in twisted.insults.__init__.py).

Either `deprecateModuleAttribute` should emit only one warning, or a different API should be provided to deprecate packages.

<details><summary>Searchable metadata</summary>

```
trac-id__4492 4492
type__defect defect
reporter__cyli cyli
priority__normal normal
milestone__ 
branch__branches_deprecated_module_4492 branches/deprecated-module-4492
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__deprecation deprecation
time__1275781656000000 1275781656000000
changetime__1279055585000000 1279055585000000
version__None None
owner__ 
cc__therve
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_1"></a>|@glyph set owner to @jml|
|-|-|

I think the deprecation module is mostly yours, jml?

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [29601]) Branching to 'deprecated-module-4492'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun removed owner|
|-|-|

[Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/deprecated-module-4492), please review.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve set owner to @exarkun|
|-|-|

Looks great, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [29817]) Merge deprecated-module-4492

Author: exarkun
Reviewer: therve
Fixes: [#4492](https://github.com/twisted/twisted/issues/4492)

Fix a bug in `twisted.python.deprecate.deprecateModuleAttribute` which
caused it to sometimes emit a deprecation warning twice.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
